### PR TITLE
DOC-133 | Search highlighting with ArangoSearch

### PR DESCRIPTION
**Pull request by Simran-B**
Covered by this PR:
- Reference docs
- Release notes
- Basic example using token search

Missing (separate PR?):
- ~Phrase search~ and fuzzy search examples
- Example of limiting the number of matches

TODO:
- ~Term search includes Levenshtein and others, need to reword description~
- Does it work for non-text?

### CODE REVIEW

File: 3.10/analyzers.md [hunk 1]
```diff
@@ -169,11 +169,14 @@ Currently the following *features* are supported:
...
   for `PHRASE()` and `NGRAM_MATCH()`.
   If present, then the `frequency` feature is also required.
+- **offset**: enable search highlighting capabilities (Enterprise Edition only).
+  Required for `OFFSET_INFO()`.
 
 Also see [PHRASE()](aql/functions-arangosearch.html#phrase),
 [BM25()](aql/functions-arangosearch.html#bm25),
 [TFIDF()](aql/functions-arangosearch.html#tfidf),
-[NGRAM_MATCH()](aql/functions-arangosearch.html#ngram_match).
+[NGRAM_MATCH()](aql/functions-arangosearch.html#ngram_match),
```
Comment (gnusi) [inline 3.10/analyzers.md:178]: `NGRAM_MATCH` and `PHRASE` and filtering functions, let's don't mix them up with `OFFSET_INFO`, `BM25`, `TFIDF`.

File: 3.10/aql/functions-arangosearch.md [hunk 2]
```diff
@@ -1161,3 +1163,84 @@ FOR doc IN viewName
...
+Search Highlighting Functions
+-----------------------------
+
+{% include hint-ee.md feature="Search highlighting" %}
+
+### OFFSET_INFO()
+
+`OFFSET_INFO(doc, paths) → offsetInfo`
+
+Returns the attribute paths and substring offsets of matched tokens, phrases, or
```
Comment (gnusi) [inline 3.10/aql/functions-arangosearch.md:1176]: `tokens` -> `terms`

File: 3.10/aql/functions-arangosearch.md [hunk 3]
```diff
@@ -1161,3 +1163,84 @@ FOR doc IN viewName
...
 ```
+
+Search Highlighting Functions
+-----------------------------
+
+{% include hint-ee.md feature="Search highlighting" %}
+
+### OFFSET_INFO()
+
+`OFFSET_INFO(doc, paths) → offsetInfo`
```
Comment (gnusi) [inline 3.10/aql/functions-arangosearch.md:1174]: `paths` can also be just a string, then it'll be treated the same as an array of only 1 string

File: 3.10/aql/functions-arangosearch.md [hunk 4]
```diff
@@ -1161,3 +1163,84 @@ FOR doc IN viewName
...
+
+Returns the attribute paths and substring offsets of matched tokens, phrases, or
+_n_-grams for search highlighting purposes.
+
+- **doc** (document): must be emitted by `FOR ... IN viewName`
+- **paths** (array): an array of strings, each describing an attribute and array
```
Comment (gnusi) [inline 3.10/aql/functions-arangosearch.md:1180]: let's mention that with array of string you get default offset limits

File: 3.10/aql/functions-arangosearch.md (hunk 3 above)
Comment (Simran-B) [reply 3.10/aql/functions-arangosearch.md:1174]: Hm, okay. Too many syntax variants tend to be confusing, though.

**Comment by Simran-B** [comment]
Remarks added about OFFSET_INFO() returning offsets in bytes, not characters in https://github.com/arangodb/docs/pull/1094/commits/aa40ace665c2468cc315f457837931d730bd4584

Reference: https://arangodb.atlassian.net/browse/DOC-177

**Pull request merged**